mypy/dmypy_server.py

```diff
@@ -601,6 +626,13 @@ def fine_grained_increment_follow_imports(self, sources: list[BuildSource]) -> l
         changed, new_files = self.find_reachable_changed_modules(
             sources, graph, seen, changed_paths
         )
+        if explicit_export_types:
+            # Same as in fine_grained_increment().
+            changed += [
+                (bs.module, bs.path)
+                for bs in sources
+                if bs.path and (bs.module, bs.path) not in changed
```

Comment:
Similar to above. Maybe move this to a helper function to avoid code duplication?

Reply:
Moved to a helper function.